src/Swarm/Game/Scenario/Objective/Presentation/Render.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- Display logic for Objectives.
+module Swarm.Game.Scenario.Objective.Presentation.Render where
+
+import Brick hiding (Direction, Location)
+import Brick.Widgets.Center
+import Brick.Widgets.List qualified as BL
+import Control.Lens hiding (Const, from)
+import Data.List.NonEmpty qualified as NE
+import Data.Map.Strict qualified as M
+import Data.Maybe (listToMaybe)
+import Data.Text (Text)
+import Data.Text qualified as T
+import Data.Vector qualified as V
+import Swarm.Game.Scenario.Objective
+import Swarm.Game.Scenario.Objective.Presentation.Model
+import Swarm.TUI.Attr
+import Swarm.TUI.Model.Name
+import Swarm.TUI.View.Util
+
+makeListWidget :: GoalTracking -> BL.List Name GoalEntry
+makeListWidget (GoalTracking _announcements categorizedObjs) =
+  BL.listMoveTo 1 $ BL.list ObjectivesList (V.fromList objList) 1
+ where
+  objList = concatMap f $ M.toList categorizedObjs
+  f (h, xs) = Header h : map (Goal h) (NE.toList xs)
+
+renderGoalsDisplay :: GoalDisplay -> Widget Name
+renderGoalsDisplay gd =
+  padAll 1 $
+    hBox
+      [ hLimitPercent 30 $
```

Comment:
I guess here is where we'd need to modify the code to omit the left-30% goal list when there is only a single goal.  Seems like it shouldn't be too difficult (though I admit I haven't thought all the way through the details).

Reply:
Done.  Test with:

    ./scripts/play.sh --scenario scenarios/Tutorials/backstory.yaml